packages/composable-controller/src/ComposableController.ts

```diff
@@ -126,11 +127,14 @@ export type ComposableControllerMessenger<
  * Controller that composes multiple child controllers and maintains up-to-date composed state.
  *
  * @template ComposableControllerState - A type object containing the names and state types of the child controllers.
- * @template ChildControllers - A union type of the child controllers being used to instantiate the {@link ComposableController}.
+ * @template ChildControllers - A type object that specifies the child controllers which are used to instantiate the {@link ComposableController}.
  */
 export class ComposableController<
   ComposableControllerState extends LegacyComposableControllerStateConstraint,
-  ChildControllers extends ControllerInstance,
+  ChildControllers extends Record<
```

Comment:
Nit: Is it worth renaming this type parameter to indicate that controllers are keyed by name, e.g., `ChildControllersByName`?

Reply:
How would you feel about `ChildControllersMap`? https://github.com/MetaMask/core/pull/4968/commits/db8147ae1880aa52e8b71061bff2271a4ae12744

Reply:
Okay, yeah, that works too!